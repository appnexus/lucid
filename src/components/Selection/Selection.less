@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

.lucid-Selection {
	display: flex;

	min-height: @size-standard-height;
	font: @font;
	color: @color-darkGray;
	background-color: @color-white;
	border-radius: @size-borderRadius;
	border: 1px solid @featured-color-primary-borderColorLite;
	padding-left: @size-S;
	margin-bottom: @size-XS;

	&-is-small {
		min-height: @size-L;
		font-size: @size-font-L;
	}

	&-is-bold {
		.bold();
	}

	&-has-background {
		background-color: @color-backgroundColor;
	}

	// Kinds

	&-success {
		border-color: @featured-color-success-borderColorLite;
		padding-left: @size-XS;
	}

	&-danger {
		border-color: @featured-color-danger-borderColorLite;
		padding-left: @size-XS;
	}

	&-info {
		border-color: @featured-color-info-borderColorLite;
		padding-left: @size-XS;
	}

	&-warning {
		border-color: @featured-color-warning-borderColorLite;
		padding-left: @size-XS;
	}

	&-container {
		border-color: @color-borderColorLight;
	}

	// Others

	&-Icon {
		margin-top: 5px;
		margin-right: @size-XS;
	}
	&-Icon-is-small {
		margin-top: 13px;
	}

	&-label-container {
		flex: 1;
		display: flex;
	}

	&-label {
		flex: 1;
		margin: @size-XS 0;
	}
	&-label-is-small {
		margin: 13px 0;
	}

	&-close-button {
		fill: @color-mediumGray;

		&:hover {
			fill: @color-darkGray;
		}
	}

	&-content {
		flex: 1;
	}

	&-children-container {
		padding-top: 2px;

		& > .lucid-Selection {
			margin-right: @size-XS;
			margin-bottom: @size-XXS;
		}

		& > .lucid-Selection:last-of-type {
			margin-bottom: @size-S;
		}
	}
}
