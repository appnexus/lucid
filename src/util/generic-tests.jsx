import sinon from 'sinon';
import React from 'react';
import { mount, shallow } from 'enzyme';
import assert from 'assert';
import describeWithDOM  from './describe-with-dom';
import _ from 'lodash';
import * as lucid from '../index';

// Common tests for all our components
export function common(Component, {
	getDefaultProps = _.noop,
	selector = null,
	exemptFunctionProps = [],
} = {}) {

	function generateDefaultProps(props={}) {
		return _.assign({}, getDefaultProps(), props);
	}

	describe('[common]', () => {
		it('should have a `displayName` defined', () => {
			assert(Component.displayName);
		});

		it('should pass through styles to the root element', () => {
			let style = {
				backgroundColor: '#f0f',
			};
			const wrapper = shallow(<Component {...generateDefaultProps()} style={style}/>);
			const rootWrapper = selector ? wrapper.find(selector).first() : wrapper.first();
			const rootStyle = rootWrapper.prop('style');
			assert(_.every(style, (val, key) => val === rootStyle[key]), 'root style must contain passed styles');
		});

		it('should pass through `className`', () => {
			const expectedClass = 'rAnDoM';
			const wrapper = shallow(<Component {...generateDefaultProps()} className={expectedClass}/>);
			const rootWrapper = selector ? wrapper.find(selector).first() : wrapper.first();
			const classNames = rootWrapper.prop('className').split(' ');

			assert(_.includes(classNames, expectedClass), `'${classNames}' should include '${expectedClass}'`);
		});

		it('should have an application scoped base class', () => {
			const expectedClass = 'lucid-' + Component.displayName;
			const wrapper = shallow(<Component {...generateDefaultProps()} />);
			const rootWrapper = selector ? wrapper.find(selector).first() : wrapper.first();
			const classNames = rootWrapper.prop('className').split(' ');

			assert(_.includes(classNames, expectedClass), `'${classNames}' should include '${Component.displayName}'`);
		});

		it('should have only application scoped classes', () => {
			const wrapper = shallow(<Component {...generateDefaultProps()} />);
			const rootWrapper = selector ? wrapper.find(selector).first() : wrapper.first();
			const parentClasses = rootWrapper.prop('className').split(' ');
			const childrenClasses = rootWrapper.children().reduce((acc, node) => {
				if (!node.prop('className')) {
					return acc;
				}

				return acc.concat(node.prop('className').split(' '))
			}, []);

			const allClasses = parentClasses.concat(childrenClasses);

			_.forEach(allClasses, className => {
				assert(_.includes(className, `lucid-${Component.displayName}`), `${className} must be scoped`);
			});
		});

		it('should only use onX convention for function proptypes', () => {

			_.forEach(Component.propTypes, (value, key) => {
				// See the following tests from the React source code to figure out how
				// this works: https://github.com/facebook/react/blob/v0.14.7/src/isomorphic/classic/types/__tests__/ReactPropTypes-test.js
				let props = {};
				props[key] = _.noop;

				// Here we test the validation against a noop function, if it doesn't
				// have an error, then it's probably a function. The other possible is
				// that it's an `any`, so we'll test for that below
				const isProbablyFunction = !(value(props, key) instanceof Error);

				// We'll also test out a string to see if that valid
				props[key] = '';

				const isAny = isProbablyFunction && !(value(props, key) instanceof Error);

				// If it's probably a function, and it's not `any`, then we make sure
				// it starts with `on`

				if (isProbablyFunction) {
					assert(isAny || _.startsWith(key, 'on') || _.includes(exemptFunctionProps, key), `${key} must follow onX convention`);
				}

			});

		});

		// Only run this test if it's a public component
		if (!Component._lucidIsPrivate) {
			it('should be available as an exported module from index.js', () => {
				assert(lucid[Component.displayName]);
			});
		}
	});
}

// Common tests for all our icon components
export function icons(Component) {
	describeWithDOM('[icon]', () => {
		it('should pass through isBadge prop to underlying Icon component', () => {
			const wrapper = mount(<Component isBadge={true} />);
			const classNames = wrapper.find('svg').prop('className').split(' ');
			const targetClassName = 'lucid-Icon-is-badge';
			assert(_.includes(classNames, targetClassName), `'${classNames}' should include '${targetClassName}'`);
		});
	});
}

// Common tests for all control components
export function controls(Component, { callbackName, controlSelector , eventType }) {
	// Use DOM tests here since some of our controls use dom events under the hood
	describeWithDOM('[control]', () => {
		it('should callback with `event` and `props`', () => {
			const expectedSpecialProp = 32;
			const props = {
				specialProp: expectedSpecialProp,
				[callbackName]: sinon.spy(),
			};
			const wrapper = mount(<Component {...props} />);

			wrapper.find(controlSelector).simulate(eventType);

			// Last argument should be an object with `uniqueId` and `event`
			const { props: { specialProp }, event } = _.last(props[callbackName].args[0]);

			assert(event, 'missing event');
			assert.equal(specialProp, expectedSpecialProp, 'incorrect or missing specialProp');
		});
	});
}
