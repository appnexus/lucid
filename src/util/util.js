import _ from 'lodash';
import React from 'react';

export const isDevMode = (function isReactInDev() {
	try {
		React.createClass({});
	} catch(e) {
		if (e.message.indexOf('render') !== 0) {
			return true;  // A nice, specific error message
		}
	}

	return false;  // should never happen, but play it safe.
})();

export const isNode = typeof process === 'object' && process.title === 'node';

export const logger = (function() {
	return checkIsDev() ? {
		log,
		warn,
		error
	} : {
		log: _.noop,
		warn: _.noop,
		error: _.noop
	};
})();

export function checkIsDev() {
	return isDevMode &&
		typeof window !== 'undefined' &&
		!isNode &&
		typeof console !== 'undefined';
}

function log(...args) {

	console.log(...args);

	try {
		// --- Welcome to debugging Lucid ---
		// This error was thrown as a convenience so that you can use this
		// stack to find the callsite that caused this warning to fire.
		throw new Error(args[0]);
	} catch (x) {}

}

function warn(...args) {

	console.warn(...args);

	try {
		// --- Welcome to debugging Lucid ---
		// This error was thrown as a convenience so that you can use this
		// stack to find the callsite that caused this warning to fire.
		throw new Error(args[0]);
	} catch (x) {}

}

function error(...args) {

	console.error(...args);

	try {
		// --- Welcome to debugging Lucid ---
		// This error was thrown as a convenience so that you can use this
		// stack to find the callsite that caused this warning to fire.
		throw new Error(args[0]);
	} catch (x) {}

}
