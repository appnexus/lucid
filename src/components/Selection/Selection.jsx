import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import MinusCircleIcon from '../Icon/MinusCircleIcon/MinusCircleIcon';
import SuccessIcon from '../Icon/SuccessIcon/SuccessIcon';
import CrossIcon from '../Icon/CrossIcon/CrossIcon';
import InfoIcon from '../Icon/InfoIcon/InfoIcon';
import WarningIcon from '../Icon/WarningIcon/WarningIcon';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	omitProps,
	getFirst,
	filterTypes,
	rejectTypes,
} from '../../util/component-types';

const { createElement } = React;

const { bool, func, string, node, oneOf } = PropTypes;

const cx = lucidClassNames.bind('&-Selection');

const responsiveMap = {
	small: 'small',
	medium: 'small',
	large: 'large',
};

function defaultIcon(kind, responsiveMode) {
	return kind === 'default'
		? null
		: kind === 'container'
				? null
				: kind === 'success'
						? <SuccessIcon
								className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)}
							/>
						: kind === 'danger'
								? <MinusCircleIcon
										className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)}
									/>
								: kind === 'info'
										? <InfoIcon
												className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)}
											/>
										: kind === 'warning'
												? <WarningIcon
														className={cx(
															'&-Icon',
															`&-Icon-is-${responsiveMode}`
														)}
													/>
												: null;
}

/**
 *
 * {"categories": ["communication"]}
 *
 * Used to indicate selections. It's very similar to `Tag` but is meant to be
 * used in areas of the UI that have more space available to them.
 */
const Selection = createReactClass({
	displayName: 'Selection',

	components: {
		Label: createReactClass({
			displayName: 'Selection.Label',
			propName: 'Label',
		}),

		Icon: createReactClass({
			displayName: 'Selection.Icon',
			propName: 'Icon',
		}),
	},

	propTypes: {
		/**
		 * Appended to the component-specific class names set on the root element.
		 */
		className: string,
		/**
		 * Applies an icon and styles for the kind of selection.
		 */
		kind: oneOf([
			'default',
			'container',
			'success',
			'danger',
			'info',
			'warning',
		]),
		/**
		 * Shows or hides the little "x" for a given item.
		 */
		isRemovable: bool,
		/**
		 * Gives the selection a background. This is desirable when you only have
		 * one level of nested selections.
		 */
		hasBackground: bool,
		/**
		 * Make the content text bold. This is desirable when you only have
		 * one level of nested selections.
		 */
		isBold: bool,
		/**
		 * Called when the close button is clicked.
		 *
		 * Signature: `({ props, event }) => {}`
		 */
		onRemove: func,
		/**
		 * Label of the component.
		 */
		Label: node,
		/**
		 * Display a custom icon for the selection. Generally you shouldn't need
		 * this prop since the `kind` prop will pick the correct icon for you.
		 */
		Icon: node,
		/**
		 * Arbitrary children.
		 */
		children: node,
		/**
		 * Adjusts the display of this component. This should typically be driven
		 * by screen size. Currently `small` and `large` are explicitly handled
		 * by this component.
		 */
		responsiveMode: oneOf(['small', 'medium', 'large']),
	},

	getDefaultProps() {
		return {
			kind: 'default',
			isRemovable: true,
			onRemove: _.noop,
			hasBackground: false,
			isBold: false,
			responsiveMode: 'large',
		};
	},

	handleRemove(event) {
		this.props.onRemove({ props: this.props, event });
	},

	render() {
		const {
			className,
			kind,
			isRemovable,
			children,
			hasBackground,
			isBold,
			responsiveMode: responsiveModeInput,
			...passThroughs
		} = this.props;

		const responsiveMode = responsiveMap[responsiveModeInput];
		const isSmall = responsiveMode === 'small';

		const selectionChildren = filterTypes(children, Selection);
		const otherChildren = rejectTypes(children, Selection);
		const labelProps = _.get(
			getFirst(this.props, Selection.Label),
			'props',
			{}
		);
		const iconElement = getFirst(this.props, Selection.Icon);
		const iconChildren = _.get(iconElement, 'props.children');
		const icon = iconChildren
			? createElement(iconChildren.type, {
					...iconChildren.props,
					className: cx('&-Icon', iconChildren.props.className),
				})
			: defaultIcon(kind, responsiveMode);

		return (
			<div
				{...omitProps(passThroughs, Selection)}
				className={cx(
					'&',
					`&-is-${responsiveMode}`,
					kind && `&-${kind}`,
					{
						'&-has-background': hasBackground,
						'&-is-bold': isBold,
					},
					className
				)}
			>
				{icon}

				<div className={cx('&-content')}>
					<div className={cx('&-label-container')}>
						<span
							{...labelProps}
							className={cx('&-label', isSmall && '&-label-is-small')}
						/>

						{isRemovable
							? <CrossIcon
									isClickable
									size={isSmall ? 44 : 26}
									viewBox={isSmall ? '-6 -6 28 28' : '-3 -2 20 20'}
									className={cx('&-close-button')}
									onClick={this.handleRemove}
								/>
							: null}
					</div>

					{_.isEmpty(selectionChildren)
						? null
						: <div className={cx('&-children-container')}>
								{_.map(selectionChildren, ({ props }, i) => (
									<Selection
										key={
											_.get(
												getFirst(props, Selection.Label),
												['props', 'children'],
												{}
											) + i
										}
										{...props}
									/>
								))}
							</div>}
					{otherChildren}

				</div>
			</div>
		);
	},
});

export default Selection;
