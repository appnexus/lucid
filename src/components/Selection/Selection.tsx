import _ from 'lodash';
import React from 'react';
import PropTypes from 'react-peek/prop-types';
import MinusCircleIcon from '../Icon/MinusCircleIcon/MinusCircleIcon';
import SuccessIcon from '../Icon/SuccessIcon/SuccessIcon';
import CloseIcon from '../Icon/CloseIcon/CloseIcon';
import InfoIcon from '../Icon/InfoIcon/InfoIcon';
import WarningIcon from '../Icon/WarningIcon/WarningIcon';
import { lucidClassNames } from '../../util/style-helpers';
import {
	omitProps,
	getFirst,
	FC,
	StandardProps,
} from '../../util/component-types';

const { createElement } = React;

const { bool, func, string, node, oneOf } = PropTypes;

const cx = lucidClassNames.bind('&-Selection');

function defaultIcon(
	kind: SelectionKind,
	responsiveMode?: SelectionResponsiveMode
) {
	return kind === 'default' ? null : kind === 'container' ? null : kind ===
	  'success' ? (
		<SuccessIcon className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)} />
	) : kind === 'danger' ? (
		<MinusCircleIcon className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)} />
	) : kind === 'info' ? (
		<InfoIcon className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)} />
	) : kind === 'warning' ? (
		<WarningIcon className={cx('&-Icon', `&-Icon-is-${responsiveMode}`)} />
	) : null;
}

export interface ISelectionIconProps extends StandardProps {}

const SelectionIcon: FC<ISelectionIconProps> = (): null => null;
SelectionIcon.peek = {
	description: `
        Icon that is displayed within the Selection. Any of the lucid \`*Icon\` components should work.
    `,
};
SelectionIcon.displayName = 'Selection.Icon';
SelectionIcon.propName = 'Icon';

export interface ISelectionLabelProps extends StandardProps {}

const SelectionLabel: FC<ISelectionLabelProps> = (): null => null;
SelectionLabel.peek = {
	description: `
        Label for the Selection.
    `,
};
SelectionLabel.displayName = 'Selection.Label';
SelectionLabel.propName = 'Label';

type SelectionKind =
	| 'default'
	| 'container'
	| 'success'
	| 'danger'
	| 'info'
	| 'warning';

type SelectionResponsiveMode = 'small' | 'medium' | 'large';

export interface ISelectionProps
	extends StandardProps,
		React.DetailedHTMLProps<
			React.HTMLAttributes<HTMLDivElement>,
			HTMLDivElement
		> {
	/** Applies an icon and styles for the kind of selection. */
	kind: SelectionKind;

	/** Apply to the top of a nested sequence of Selection components.
	 * Adds some spacing for a list of top level Selections with nested Selctions inside each.
	 * */
	isTop?: boolean;

	/** Only applies to \`container\` Selection components.
	 * Fills with a darker gray background.
	 * Defaults to false.
	 * */
	isFilled?: boolean;

	/** Shows or hides the little "x" for a given item. */
	isRemovable: boolean;

	/** Called when the close button is clicked. */
	onRemove: ({
		props,
		event,
	}: {
		props: ISelectionProps;
		event: React.MouseEvent;
	}) => void;

	/** Gives the selection a background. This is desirable when you only have
	 * one level of nested selections.
	 * */
	hasBackground: boolean;

	/** Make the content text bold. This is desirable when you only have one
	 * level of nested selections.
	 * */
	isBold: boolean;

	/** Label of the component. */
	Label?: React.ReactNode;

	/** Display a custom icon for the selection. Generally you shouldn't need
	 * this prop since the \`kind\` prop will pick the correct icon for you.
	 * */
	Icon?: React.ReactNode;

	/** Adjusts the display of this component. This should typically be driven by
	 * screen size. Currently \`small\` and \`large\` are explicitly handled by
	 * this component.
	 * */
	responsiveMode: SelectionResponsiveMode;
}

const defaultProps = {
	isRemovable: true,
	onRemove: _.noop,
	hasBackground: false,
	isBold: false,
	kind: 'default' as const,
	responsiveMode: 'large' as const,
};

const Selection = (props: ISelectionProps) => {
	const {
		className,
		isRemovable,
		children,
		hasBackground,
		isBold,
		isFilled,
		isTop,
		kind,
		onRemove,
		responsiveMode,
		...passThroughs
	} = props;

	const isSmall = responsiveMode === 'small';

	const labelProps = _.get(getFirst(props, Selection.Label), 'props', {});
	const iconElement = getFirst(props, Selection.Icon);
	const iconChildren = _.get(iconElement, 'props.children');
	const icon = iconChildren
		? createElement(iconChildren.type, {
				...iconChildren.props,
				className: cx('&-Icon', iconChildren.props.className),
		  })
		: defaultIcon(kind, responsiveMode);

	return (
		<div
			{...omitProps(passThroughs, undefined, _.keys(Selection.propTypes))}
			className={cx(
				'&',
				`&-is-${responsiveMode}`,
				kind && `&-${kind}`,
				{
					'&-has-background': hasBackground,
					'&-is-bold': isBold,
					'&-is-filled': isFilled,
					'&-is-top': isTop,
					'&-no-title': _.isEmpty(labelProps),
				},
				className
			)}
		>
			{icon}

			<div className={cx('&-content')}>
				<div className={cx('&-label-container')}>
					<span
						{...labelProps}
						className={cx('&-label', isSmall && '&-label-is-small')}
					/>

					{isRemovable ? (
						<CloseIcon
							isClickable
							size={!isSmall ? 8 : 16}
							className={cx(
								'&-close-button',
								isSmall && '&-close-button-is-small'
							)}
							onClick={({ event }) => {
								onRemove({ event, props });
							}}
						/>
					) : null}
				</div>
				{!_.isEmpty(children) && (
					<div className={cx('&-children-container')}>
						{_.map(React.Children.toArray(children), (child, i) => {
							if (React.isValidElement(child) && child.type === Selection) {
								return (
									<Selection
										key={
											_.get(
												getFirst(child.props, Selection.Label),
												['props', 'children'],
												{}
											) + i
										}
										{...child.props}
									/>
								);
							}
							return child;
						})}
					</div>
				)}
			</div>
		</div>
	);
};

Selection.displayName = 'Selection';
Selection.Icon = SelectionIcon;
Selection.Label = SelectionLabel;
Selection.peek = {
	description: `
                Used to indicate selections. Selection is very similar to \`Tag\` but is meant
                to be used in areas of the UI that have more space available to them.
            `,
	categories: ['communication'],
};
Selection.defaultProps = defaultProps;
Selection.propTypes = {
	className: string`
			Appended to the component-specific class names set on the root element.
		`,

	kind: oneOf(['default', 'container', 'success', 'danger', 'info', 'warning'])`
			Applies an icon and styles for the kind of selection.
		`,

	isTop: bool`
			Apply to the top of a nested sequence of Selection components.
			Adds some spacing for a list of top level Selections with nested Selctions inside each.
		`,

	isFilled: bool`
			Only applies to \`container\` Selection components.
			Fills with a darker gray background.
			Defaults to false.
		`,

	isRemovable: bool`
			Shows or hides the little "x" for a given item.
		`,

	hasBackground: bool`
			Gives the selection a background. This is desirable when you only have
			one level of nested selections.
		`,

	isBold: bool`
			Make the content text bold. This is desirable when you only have one
			level of nested selections.
		`,

	onRemove: func`
			Called when the close button is clicked.
		`,

	Label: node`
			Label of the component.
		`,

	Icon: node`
			Display a custom icon for the selection. Generally you shouldn't need
			this prop since the \`kind\` prop will pick the correct icon for you.
		`,

	children: node`
			Arbitrary children.
		`,

	responsiveMode: oneOf(['small', 'medium', 'large'])`
			Adjusts the display of this component. This should typically be driven by
			screen size. Currently \`small\` and \`large\` are explicitly handled by
			this component.
		`,
};

export default Selection;
