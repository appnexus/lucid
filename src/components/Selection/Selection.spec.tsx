import React from 'react';
import { shallow } from 'enzyme';
import { common } from '../../util/generic-tests';
import Selection from './Selection';

describe('Selection', () => {
	common(Selection);

	describe('render', () => {
		it('should match snapshot for responsiveMode small', () => {
			const wrapper = shallow(
				<Selection responsiveMode='small'>Yolo</Selection>
			);

			expect(wrapper).toMatchSnapshot();
		});

		it('should match snapshot for container', () => {
			expect(
				shallow(<Selection kind='container'>cont</Selection>)
			).toMatchSnapshot();
		});

		it('should match snapshot for success', () => {
			expect(
				shallow(<Selection kind='success'>cont</Selection>)
			).toMatchSnapshot();
		});

		it('should match snapshot for danger', () => {
			expect(
				shallow(<Selection kind='danger'>cont</Selection>)
			).toMatchSnapshot();
		});

		it('should match snapshot for info', () => {
			expect(
				shallow(<Selection kind='info'>cont</Selection>)
			).toMatchSnapshot();
		});

		it('should match snapshot for warning', () => {
			expect(
				shallow(<Selection kind='warning'>cont</Selection>)
			).toMatchSnapshot();
		});

		it('should match snapshot for non removable', () => {
			expect(shallow(<Selection isRemovable={false} />)).toMatchSnapshot();
		});

		it('should match snapshot for nested selections', () => {
			const wrapper = shallow(
				<Selection>
					Hello
					<Selection>There</Selection>
				</Selection>
			);

			expect(wrapper).toMatchSnapshot();
		});

		it('should match snapshot for custom icons', () => {
			const wrapper = shallow(
				<Selection>
					<Selection.Icon>
						<div className='fake-icon' />
					</Selection.Icon>
				</Selection>
			);

			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('props', () => {
		it('onRemove', () => {
			const onRemove = jest.fn();
			const wrapper = shallow(<Selection onRemove={onRemove} />);

			wrapper.find('CloseIcon').prop('onClick')({});
			expect(onRemove).toHaveBeenCalled();
		});
	});
});
