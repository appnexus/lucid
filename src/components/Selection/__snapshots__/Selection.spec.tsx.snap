// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection render should match snapshot for container 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-container lucid-Selection-no-title"
>
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      cont
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for custom icons 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-default lucid-Selection-no-title"
>
  <div
    className="lucid-Selection-Icon fake-icon"
  />
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      <Selection.Icon
        key=".0"
      >
        <div
          className="fake-icon"
        />
      </Selection.Icon>
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for danger 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-danger lucid-Selection-no-title"
>
  <MinusCircleIcon
    aspectRatio="xMidYMid meet"
    className="lucid-Selection-Icon lucid-Selection-Icon-is-large"
    color="primary"
    isClickable={false}
    isDisabled={false}
    onClick={[Function]}
    onSelect={[Function]}
    size={16}
    viewBox="0 0 16 16"
  />
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      cont
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for info 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-info lucid-Selection-no-title"
>
  <InfoIcon
    aspectRatio="xMidYMid meet"
    className="lucid-Selection-Icon lucid-Selection-Icon-is-large"
    color="primary"
    isClickable={false}
    isDisabled={false}
    onClick={[Function]}
    onSelect={[Function]}
    size={16}
    viewBox="0 0 16 16"
  />
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      cont
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for nested selections 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-default lucid-Selection-no-title"
>
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      Hello
      <Selection
        hasBackground={false}
        isBold={false}
        isRemovable={true}
        key="[object Object]1"
        kind="default"
        onRemove={[Function]}
        responsiveMode="large"
      >
        There
      </Selection>
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for non removable 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-default lucid-Selection-no-title"
>
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for responsiveMode small 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-small lucid-Selection-default lucid-Selection-no-title"
>
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label lucid-Selection-label-is-small"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button lucid-Selection-close-button-is-small"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={16}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      Yolo
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for success 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-success lucid-Selection-no-title"
>
  <SuccessIcon
    aspectRatio="xMidYMid meet"
    className="lucid-Selection-Icon lucid-Selection-Icon-is-large"
    color="primary"
    isClickable={false}
    isDisabled={false}
    onClick={[Function]}
    onSelect={[Function]}
    size={16}
    viewBox="0 0 16 16"
  />
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      cont
    </div>
  </div>
</div>
`;

exports[`Selection render should match snapshot for warning 1`] = `
<div
  className="lucid-Selection lucid-Selection-is-large lucid-Selection-warning lucid-Selection-no-title"
>
  <WarningIcon
    aspectRatio="xMidYMid meet"
    className="lucid-Selection-Icon lucid-Selection-Icon-is-large"
    color="primary"
    isClickable={false}
    isDisabled={false}
    onClick={[Function]}
    onSelect={[Function]}
    size={16}
    viewBox="0 0 16 16"
  />
  <div
    className="lucid-Selection-content"
  >
    <div
      className="lucid-Selection-label-container"
    >
      <span
        className="lucid-Selection-label"
      />
      <CloseIcon
        aspectRatio="xMidYMid meet"
        className="lucid-Selection-close-button"
        color="primary"
        isClickable={true}
        isDisabled={false}
        onClick={[Function]}
        onSelect={[Function]}
        size={8}
        viewBox="0 0 16 16"
      />
    </div>
    <div
      className="lucid-Selection-children-container"
    >
      cont
    </div>
  </div>
</div>
`;
